presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -55,10 +55,14 @@ public PickTableLayout(Metadata metadata)
             implements Rule<FilterNode>
     {
         private final Metadata metadata;
+        private final SqlParser sqlParser;
+        private final boolean replaceExistingLayout;
 
-        private PickTableLayoutForPredicate(Metadata metadata)
+        private PickTableLayoutForPredicate(Metadata metadata, SqlParser sqlParser, boolean replaceExistingLayout)
         {
             this.metadata = requireNonNull(metadata, "metadata is null");
+            this.replaceExistingLayout = replaceExistingLayout;
+            this.sqlParser = sqlParser;
```

Comment:
use require non null, here and always in constructor (unless it is some exceptional case)